### PR TITLE
Button Editor: Fix keypress events going to the editor when the buttons are focused

### PR DESCRIPTION
**Pull request by EchterAgo**

### CODE REVIEW

File: gui/qt/paytoedit.py [hunk 1]
```diff
@@ -266,23 +266,24 @@ def textUnderCursor(self):
...
+        if e.key() in [Qt.Key_Tab, Qt.Key_Backtab]:
             e.ignore()
             return
 
         if e.key() in [Qt.Key_Down, Qt.Key_Up] and not self.is_multiline():
             e.ignore()
             return
 
-        QPlainTextEdit.keyPressEvent(self, e)
+        super().keyPressEvent(e)
```
Comment (cculianu) [inline gui/qt/paytoedit.py]: lol. Yeah. 👍 

Esp. now that we do base class magic here.  Good catch.

File: gui/qt/paytoedit.py [hunk 2]
```diff
@@ -266,23 +266,24 @@ def textUnderCursor(self):
         return tc.selectedText().strip()
 
     def keyPressEvent(self, e):
-        if self.isReadOnly():
+        if self.isReadOnly() or not self.hasFocus():
+            e.ignore()
```
Comment (cculianu) [inline gui/qt/paytoedit.py]: Just for my own edification: it's possible to receive the even even if not hasFocus() if a child widget ignores the event right? I seem to remember that's how it works hence how we ended up here...

File: gui/qt/paytoedit.py (hunk 1 above)
Comment (cculianu) [reply gui/qt/paytoedit.py]: Also -- do we want to check if e.isIgnored() here (or whatever the method is called in Qt)? Or? I forget what the good well-behaved contract should be. Just asking...

File: gui/qt/paytoedit.py (hunk 2 above)
Comment (EchterAgo) [reply gui/qt/paytoedit.py]: Yea, that is my understanding as well. Note that you have to explicitly call .ignore() on the event in the child as accepting it is the default.

File: gui/qt/paytoedit.py (hunk 1 above)
Comment (EchterAgo) [reply gui/qt/paytoedit.py]: isAccepted would always be true, it is the default.

**Comment by EchterAgo** [comment]
Tested on Windows and Linux

File: gui/qt/paytoedit.py (hunk 1 above)
Comment (cculianu) [reply gui/qt/paytoedit.py]: ah heh

ok nevermind.. will just merge

**Pull request merged**